src/core/qgsproject.cpp

```diff
@@ -197,11 +197,13 @@ QgsProjectProperty *findKey_( const QString &scope,
 \param key key name
 \param rootProperty is the property from which to start adding
 \param value the value associated with the key
+\param propertiesModified the parameter will be set to true if the written entry modifies pre-existing properties
 */
 QgsProjectProperty *addKey_( const QString &scope,
                              const QString &key,
                              QgsProjectPropertyKey *rootProperty,
-                             const QVariant &value )
+                             const QVariant &value,
+                             bool &propertiesModified )
```

Comment:
why not use directly setDirty inside addKey_ when necessary?

Reply:
@luipir , because we'd have to pass on a QgsProject pointer to do that, which doesn't seem to be a good idea to begin with :)

Reply:
has ok, I supposed addKey_ was a private QgsProject member, sorry for the noise